batch7crud-roll9/src/main/java/com/lftechnology/batch7crud/dao/StudentDAO.java

```diff
@@ -0,0 +1,140 @@
+                std.setMiddleName(rs.getString(3));
+                std.setLastName(rs.getString(4));
+                std.setAddress(rs.getString(5));
+                std.setGrade(rs.getInt(6));
+                stdList.add(std);
+            }
+            return stdList;
+
+        } catch (SQLException ex) {
+            throw new DataException();
+        }
+    }
+
+    public void delete(int id) throws DataException//get the object
+    {
+        try {
+            Connection conn = DBConnection.getConnection();
+            PreparedStatement pstmt= conn.prepareStatement("DELETE FROM tbl_userinfo WHERE id=?;");
+            pstmt.setInt(1, id);
+            System.out.println(pstmt.toString());
+            pstmt.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataException();
+        }
+    }
+
+    public Student fetchById(int id) throws DataException
+    {
+        try {
+            Connection conn = DBConnection.getConnection();
+            Student std = null;
+            ResultSet rs;
+            PreparedStatement pstmt= conn.prepareStatement("SELECT * from tbl_userinfo WHERE id=?");
+            pstmt.setInt(1, id);
+            rs = pstmt.executeQuery();
+            while(rs.next()){
+                std = new Student();
+                std.setId(rs.getInt(1));
+                std.setFirstName(rs.getString(2));
+                std.setMiddleName(rs.getString(3));
+                std.setLastName(rs.getString(4));
+                std.setAddress(rs.getString(5));
+                std.setGrade(rs.getInt(6));
+            }
+            return std;
+
+        } catch (SQLException ex) {
+            throw new DataException();
+        }
+    }
+
+
+    public void edit(Student s) throws DataException
+    {
+        try {
+            Connection conn = DBConnection.getConnection();
+            ResultSet rs;
+            PreparedStatement pstmt= conn.prepareStatement("UPDATE tbl_userinfo SET firstname=?, middlename=?, lastname=?,address=?,grade=? WHERE id=?");
+            pstmt.setString(1, s.getFirstName());
+            pstmt.setString(2, s.getMiddleName());
+            pstmt.setString(3, s.getLastName());
+            pstmt.setString(4, s.getAddress());
+            pstmt.setInt(5, s.getGrade());
+            pstmt.setInt(6, s.getId());
+
+            pstmt.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataException();
+        }
+    }
+
+    public int studentCount() throws DataException//get the object
+    {
+        try {
+            int totalStudents = 0;
+            Connection conn = DBConnection.getConnection();
+            PreparedStatement pstmt = conn.prepareStatement("SELECT count(*) FROM tbl_userinfo");
+            ResultSet rs = pstmt.executeQuery();
+            while(rs.next())
+                totalStudents = rs.getInt(1);
+            return totalStudents;
+        } catch (SQLException ex) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)